src/main/java/store/model/Product.java

```diff
@@ -0,0 +1,76 @@
+package store.model;
+
+import camp.nextstep.edu.missionutils.DateTimes;
+
+import java.text.NumberFormat;
+import java.time.LocalDateTime;
+
+public class Product {
+    private final String name;
+    private int price;
+    private int quantity;
+    private Promotion promotion;
+
+    public Product(String name, int price, int quantity, Promotion promotion) {
+        this.name = name;
+        this.price = price;
+        this.quantity = quantity;
+        this.promotion = promotion;
+    }
+
+    public String getName() {
+        return name;
+    }
+
+    public int getQuantity() {
+        return quantity;
+    }
+
+    public int getPrice() {
+        return price;
+    }
+
+    public int addPromotions(int number) {
+        int buy = promotion.getBuy();
+        int get = promotion.getGet();
+        if (number % (buy + get) >= buy) {
+            return get - (number % (buy + get) - buy);
+        }
+        return 0;
+    }
+
+    public int applyPromotions(int number) {
+        int buy = promotion.getBuy();
+        int get = promotion.getGet();
+        return (number / (get + buy)) * get;
+    }
+
+    public boolean isPromotion() {
+        if (promotion == null) {
+            return false;
+        }
+        LocalDateTime time = DateTimes.now();
+        if (time.isBefore(promotion.getStartDate()) || time.isAfter(promotion.getEndDate())) {
+            return false;
+        }
+        return true;
```

Comment:
아래처럼 축약하면 어떨까요? 흠 너무 긴가요?!🤔
```suggestion
        return !time.isBefore(promotion.getStartDate()) && !time.isAfter(promotion.getEndDate());
```